File: quote/Projects/PluginModel/ShippingDB.cs
namespace Model
{
    using System;
    using System.Collections.ObjectModel;
    using System.Linq;

    public sealed class ShippingDB
    {
        private ShippingDB()
        {
        }

        public static void InitializeShipping()
        {
            DB.QuoteDataBase._ShippingDataTable table = new DB.QuoteDataBase._ShippingDataTable();
            DB.QuoteDataBaseTableAdapters._ShippingTableAdapter adapter = new DB.QuoteDataBaseTableAdapters._ShippingTableAdapter();
            adapter.Fill(table);

            string[] a = new string[table.Count + 1];
            int i = 0;
            foreach (DB.QuoteDataBase._ShippingRow s in table.Rows) 
            {
                a[i] = s.Index;
                Shipping.Dictionary.Add(s.Index, s.Cost);
                i += 1;
            }

            Shipping.SetupDescriptions(new ReadOnlyCollection<string>(a));
        }
    }
}

File: quote/Projects/DekalbProperties/ShippingDB.vb
Imports System.Collections.ObjectModel
Imports System.Linq

Public NotInheritable Class ShippingDB

    Private Sub New()
    End Sub

    Public Shared Sub Initialize()
        Dim table As New DB.QuoteDataBase._ShippingDataTable()
        Dim adapter As New DB.QuoteDataBaseTableAdapters._ShippingTableAdapter()
        adapter.Fill(table)

        Dim a As String() = New String(table.Count) {}
        Dim i As Integer = 0
        For Each s As DB.QuoteDataBase._ShippingRow In table.Rows
            a(i) = s.Index
            Shipping.Dictionary.Add(s.Index, s.Cost)
            i += 1
        Next

        Shipping.SetupDescriptions(New ReadOnlyCollection(Of String)(a))
    End Sub
End Class
